File: backend/src/main/java/br/edu/unoesc/vestock/model/Fornecedor.java
package br.edu.unoesc.vestock.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrimaryKeyJoinColumn;
import jakarta.persistence.Table;

/**
 * Representa um fornecedor.
 */
@Entity
@Table(name = "fornecedor")
@PrimaryKeyJoinColumn(name = "codpes")
public class Fornecedor extends Pessoa {

	/** CNPJ do fornecedor. */
	@Column(name = "cnpfor", length = 18, nullable = false, unique = true)
	private String cnpj;

	/** Nome fantasia do fornecedor. */
	@Column(name = "nomfanfor", length = 100)
	private String nomeFantasia;

	@ManyToOne
	@JoinColumn(name = "codloj", nullable = false)
	private Loja loja;

	public Fornecedor() {

	}

	public Fornecedor(Integer id, String nome, String email, Character sexo, String telefone, String rua, String bairro,
			String cidade, String estado, String cnpj, String nomeFantasia) {
		super(id, nome, email, sexo, telefone, rua, bairro, cidade, estado);
		this.cnpj = cnpj;
		this.nomeFantasia = nomeFantasia;
	}

	// Getters e Setters

	public String getCnpj() {
		return cnpj;
	}

	public void setCnpj(String cnpj) {
		this.cnpj = cnpj;
	}

	public String getNomeFantasia() {
		return nomeFantasia;
	}

	public void setNomeFantasia(String nomeFantasia) {
		this.nomeFantasia = nomeFantasia;
	}

	public Loja getLoja() {
		return loja;
	}

	public void setLoja(Loja loja) {
		this.loja = loja;
	}
}

File: front/lib/models/fornecedor.dart
class Fornecedor {
  final int? id, lojaId;
  final String nome, cnpj, telefone, email;
  final String? nomeFantasia, sexo, rua, bairro, cidade, estado;

  Fornecedor({
    this.id,
    this.lojaId,
    required this.nome,
    required this.cnpj,
    required this.telefone,
    required this.email,
    this.nomeFantasia,
    this.sexo,
    this.rua,
    this.bairro,
    this.cidade,
    this.estado,
  });

  factory Fornecedor.fromJson(Map<String, dynamic> j) => Fornecedor(
    id: j['id'],
    lojaId: j['loja'] != null ? j['loja']['id'] : null,
    nome: j['nome'] ?? '',
    cnpj: j['cnpj'] ?? '',
    telefone: j['telefone'] ?? '',
    email: j['email'] ?? '',
    nomeFantasia: j['nomeFantasia'],
    sexo: j['sexo'],
    rua: j['rua'],
    bairro: j['bairro'],
    cidade: j['cidade'],
    estado: j['estado'],
  );

  Map<String, dynamic> toJson() => <String, dynamic>{
    'nome': nome,
    'cnpj': cnpj,
    'telefone': telefone,
    'email': email,
    'nomeFantasia': nomeFantasia,
    'sexo': sexo,
    'rua': rua,
    'bairro': bairro,
    'cidade': cidade,
    'estado': estado,
    if (lojaId != null) 'loja': <String, int?>{'id': lojaId},
  };
}
